r3bgen/R3BBeamProperties.h

```diff
@@ -0,0 +1,59 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#ifndef R3BBEAMPROPERTIES
+#define R3BBEAMPROPERTIES
+
+#include "R3BDistribution.h"
+#include "R3BDistribution1D.h"
+
+class R3BBeamProperties
+{
+  public:
+    R3BBeamProperties();
+
+    // Vertex distribution in cm
+    R3BDistribution<3>& GetVertexDistribution() { return fVertex_cm; }
+    void SetVertexDistribution(const R3BDistribution<3>& vertexDistribution_x_y_z_cm)
+    {
+        fVertex_cm = vertexDistribution_x_y_z_cm;
+    }
+
+    // spread in mRad
+    R3BDistribution<2>& GetSpreadDistribution() { return fSpread_mRad; }
+    void SetSpreadDistribution(const R3BDistribution<2>& spreadDistribution_theta_phi_mRad)
+    {
+        fSpread_mRad = spreadDistribution_theta_phi_mRad;
+    }
+    void SetSpreadDistribution(const R3BDistribution<1>& spreadDistribution_theta_mRad)
+    {
+        fSpread_mRad =
+            R3BDistribution<2>({ spreadDistribution_theta_mRad, R3BDistribution1D::Flat(0, 2 * TMath::Pi() * 1e3) });
+    }
+
+    // beta distribution
+    R3BDistribution<1>& GetBetaDistribution() { return fBeta; }
+    void SetBetaDistribution(const R3BDistribution<1>& betaDistribution) { fBeta = betaDistribution; }
+
+    // This method will calculate the beam beta by assuming nucleon mass.
```

Comment:
```suggestion
```

Reply:
I will wait two more days in case @kresan or someone else has also other suggestions, then I will commit those changes and the changes in the macros aswell.

Reply:
Note that the test is in the main repository <sub><sup>(as it should be)</sup></sub>, not in the macro repository - to get the travis tasks passing.